packages/backend-app-api/src/config/config.ts

```diff
@@ -109,11 +111,13 @@ export async function loadBackendConfig(options: {
       }),
     },
   });
-
   console.info(
     `Loaded config from ${appConfigs.map(c => c.context).join(', ')}`,
   );
 
+  if (options.additionalConfigs) {
+    appConfigs.push(...options.additionalConfigs);
```

Comment:
Question on the usage here, do you think it's best to have this config override any existing `app-config` files, or the other way around? Right now it's overriding config files.

Reply:
I guess what's in `app-config.yaml` should override it to allow developers to hard code secrets while doing local development. What do you think?

Reply:
On second thought, the secret loading code would still run and without proper config, it will fail. I'm not really sure.

Reply:
It should only override configs if the keys collide on non-object types, otherwise to override recursively. 

The way that I had set it up previously with `mergeJson` was that for a given pair of configs such as:
`config A`
```json
{
  "name": "Alex",
  "database": {
    "provider": "mysql",
  },
  "list": [1, 2, 3]
}
```

`config B`
```json
{
  "database": {
    "url": "mysql://user:name@localhost:3206/backstage"
  },
  "list": ["a", "b", "c"]
}
```

Then the result of merged config would look like:
```ts
const mergedConfig = mergeJson(configA, configB);
// mergedConfig:
{
  "name": "Alex",
  "database": {
    "provider": "mysql",
    "url": "mysql://user:name@localhost:3206/backstage"
  },
  "list": ["a", "b", "c"]
}
```

Where the configs further on the right take precedence over those on the left. 

Is this how the `ConfigReader.fromConfigs` function currently works?

Reply:
If I understood @Rugvip correctly, I think the question is "what would the user expect to happen, and how should it work?" The original purpose of this work is to provide an escape hatch for security-conscious adopters who are not allowed to use environment variables to store secrets. Instead of exposing secrets via the environment variable to each Node module in the process, this mechanism will allow us to inject those values into the config without using environment variables. 

But what happens to the configuration in app-config.yaml? Which configuration should win?

Reply:
@Rugvip I guess with my previous comment, all I'm trying to say is that it should only override when there is a non-object collision. Otherwise it should attempt to merge objects recursively. I don't see any reason why the user should have to provide a copy of their config on file just to avoid hard-coding credentials. 

This is outside the scope of the current PR, but maybe we can introduce some form of config schema validation that users can adhere to. This way, if the end result of a config is 
```ts
type Config = {
  database: {
    provider: "mysql" | "sqlite";
    url: string;
  };
}
```

It could error out if it didn't get any of the required values at the end of merging.

Reply:
@taras The config supplied within code should be expected to be smaller but doesn't have to be. The user should expect the `additionalConfigs` provided to take priority over the configs read from disk or network.